src/rust/ide/lib/view/src/graph_editor/component/documentation_view.rs

```diff
@@ -0,0 +1,159 @@
+        let screen = scene.camera().screen();
+        let size   = Rc::new(Cell::new(Vector2(290.0,screen.height - 30.0)));
+
+        dom.dom().set_style_or_warn("white-space"     ,"normal"                             ,&logger);
+        dom.dom().set_style_or_warn("overflow-y"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("overflow-x"      ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("background-color","rgba(255, 255, 255, 0.85)"          ,&logger);
+        dom.dom().set_style_or_warn("padding"         ,"5px"                                ,&logger);
+        dom.dom().set_style_or_warn("pointer-events"  ,"auto"                               ,&logger);
+        dom.dom().set_style_or_warn("border-radius"   ,"14px"                               ,&logger);
+        dom.dom().set_style_or_warn("width"           ,format!("{}px", 290)                 ,&logger);
+        dom.dom().set_style_or_warn("height"          ,format!("{}px", screen.height - 30.0),&logger);
+
+        scene.dom.layers.main.manage(&dom);
+        DocumentationViewModel{dom,logger,size}.init()
+    }
+
+    fn init(self) -> Self {
+        self.reload_style();
+        self
+    }
+
+    fn set_size(&self, size:Vector2) {
+        self.size.set(size);
+        self.reload_style();
+    }
+
+    const PLACEHOLDER_STR: &'static str = "<h3>Enso Documentation Viewer</h3>\
+                                           <p>No documentation available</p>";
+
+    fn receive_data(&self, data:&Data) -> Result<(),DataError> {
+        let data_inner = match data {
+            Data::Json {content} => content,
+            _                    => todo!(),
+        };
+
+        let data_str = serde_json::to_string_pretty(&**data_inner);
+        let data_str = data_str.unwrap_or_else(|e| format!("<Cannot render data: {}>", e));
+        // Fixes a Doc Parser Bug - to be removed when rewritten to rust
+        let data_str = data_str.replace("\\n", "\n");
+        let data_str = data_str.replace("\"", "");
+
+        let parser = parser::DocParser::new_or_panic();
+        let output = parser.generate_html_doc_pure(data_str);
+        let output = output.unwrap_or_else(|_| String::from(DocumentationViewModel::PLACEHOLDER_STR));
+        // Fixes a Doc Parser related idea, where stylesheet was a separate file
+        let output = output.replace(r#"<link rel="stylesheet" href="style.css" />"#, "");
+
+        let data_str = format!(r#"<div class="docVis">{}{}</div>"#, get_doc_style(), output);
+        self.dom.dom().set_inner_html(&data_str);
+        Ok(())
+    }
+
+    /// Generates welcome screen HTML.
+    pub fn welcome_screen(&self) {
+        let placeholder = DocumentationViewModel::PLACEHOLDER_STR;
+        let data_str    = format!(r#"<div class="docVis">{}{}</div>"#, get_doc_style(), placeholder);
+        self.dom.dom().set_inner_html(&data_str)
+    }
```

Comment:
what is welcome screen? This needs much much better docs and descriptions.

Reply:
It is the content in the documentation view when no data has been yet received - think of it like in Xcode when nothing is opened, it says "No editor", or in right panel -  "No selection"